invite_team_members/views.py

```diff
@@ -0,0 +1,127 @@
+                verified = TeamMember.objects.get(team=team, user=user).verified
+                if verified:
+                    return Response({'error': 'User is already in the team'}, status=status.HTTP_400_BAD_REQUEST)
+                return Response({'error': 'User is already in the process of being invited to the team'},
+                                status=status.HTTP_400_BAD_REQUEST)
+            invite_token = InviteTeamMemberToken.objects.create(user=user, team=team)
+
+            team_member = TeamMember.objects.create(user=user, team=team, verified=False)
+
+            accept_url = f"{os.environ.get('FRONTEND_URL', '')}/invite-member?key={invite_token.key}"
+            email_content = f'''
+                        You have been invited to join Team {team.name}.\n
+                        Please follow below link to accept the invitation and join the team. \n\n
+                        {accept_url}
+                        This link is valid for 1 week.
+                        '''
+
+            email_content_html = render_to_string('email/accept_invite.html', {
+                'team_name': team.name,
+                'accept_url': accept_url,
+            })
+
+            send_mail(
+                f'MonAPI Invitation to Team {team.name}',
+                email_content,
+                None,
+                [user.email],
+                html_message=email_content_html,
+                fail_silently=False,
+            )
+
+            return Response({'success': True})
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class AcceptInviteView(views.APIView):
+
+    def post(self, request, format=None):
+        serializer = AcceptInviteSerializer(data=request.data)
+        if serializer.is_valid():
+            invite_token = InviteTeamMemberToken.objects.filter(
+                key=serializer.validated_data['key'],
+                created_at__gte=timezone.now() - timedelta(weeks=1)
+            )
+            if len(invite_token) == 0:
+                return Response({'error': 'Invalid token'}, status=status.HTTP_400_BAD_REQUEST)
+            invite_token = invite_token[0]
+            team_target = invite_token.team
+            user_target = invite_token.user
+
+            team_member = TeamMember.objects.get(user=user_target, team=team_target)
+            if team_member.verified:
+                return Response({'error': 'You are already a member of the team'},
+                                status=status.HTTP_400_BAD_REQUEST)
+
+            team_member.verified = True
+            team_member.save()
+            invite_token.delete()
+            return Response({'success': True})
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+class CancelInviteView(views.APIView):
+
+    def post(self, request, format=None):
+        serializer = CancelInviteSerializer(data=request.data)
+        if serializer.is_valid():
```

Comment:
ini cancelnya apakah mending pakai team id sama user id saja (?)

Reply:
Didn't think of that, changed! Now cuma verified team member yang bisa cancel invite!